server/pbench/bin/pbench-satellite-cleanup

```diff
@@ -111,20 +111,24 @@ for host in $hosts; do
             echo $TS: Failed to remove the tarball from Incoming directory: $INCOMING/$host/${x%%.tar.xz} | tee -a $mail_content >&4 
             nincomingerrs=$nincomingerrs+1
         fi
-        # remove from results
-        prefix_value=$(cat ".prefix/prefix.${x%%.tar.xz}")
-        if [ -L $RESULTS/$host/${x%%.tar.xz} ]; then
-            rm $RESULTS/$host/${x%%.tar.xz}
-            rc=$?
-            if [ $rc != 0 ]; then
-                echo $TS: Failed to remove: $RESULTS/$host/${x%%.tar.xz} | tee -a $mail_content >&4 
-                nresultserrs=$nresultserrs+1
+        #remove the results
+        tar_name=$RESULTS/$host/${x%%.tar.xz}
+        if [ -s .prefix/prefix.${x%%.tar.xz} ]; then
+            prefix_value=$(cat ".prefix/prefix.${x%%.tar.xz}")
+            prefix_tar_name=`echo $tar_name | sed 's;'${x%%.tar.xz}';'$prefix_value/${x%%.tar.xz}';'`
```

Comment:
No, just set `tar_name` to one or the the other and then have just *one* if-block: if tar_name exists then remove it and take care of any errors. No need to have two variables, tar_name and prefix_tar_name.

Reply:
Also, tar_name is a bad name: it's the link that we are getting rid of. And there is no need to sed: just concatenate the various pieces.